analysis/spectra_chisq/_model_spectra.py

```diff
@@ -0,0 +1,475 @@
+    ax2.axhline(0, color='black', linestyle=':')
+    ax2.set_xlabel('obs frame wavelength [Angstrom]', fontsize=14)
+    ax2.set_ylabel('Observed - model flux \n [ergs/(s cm^2 Angstrom)]', fontsize=14)
+
+    return None
+
+
+def plot_spectra(x1, y1, x2, y2):
+    """Plot the spectra from the data and the model (flux vs wavelength).
+    
+    Args:
+        x1 (1d array-like of floats): Observer frame wavelength
+        y1 (1d array-like of floats): Observed flux
+        x2 (1d array-like of floats): Observer frame wavelength
+        y2 (1d array-like of floats): Modeled flux
+    """
+    
+    plt.clf()
+    # Set figure size
+    plt.rcParams['figure.figsize'] = [12, 6] # units: inches
+    
+    # Plot: flux vs wavelength
+    # Plot  wavelength vs obs flux
+    plt.plot(x1, y1, linestyle=' ', marker='.', label='obs')
+    # Plot wavelength vs model flux
+    plt.plot(x2, y2, linestyle=' ', marker='.', label='fit')
+    # Labels
+    plt.xlabel('obs frame wavelength [Angstrom]')
+    plt.ylabel('flux [ergs/(s cm^2 Angstrom)]', fontsize=14)
+    plt.legend(fontsize=14)
+    #plt.suptitle('id: {}, band: {}'.format(OID,BND), fontsize=14)
+    # Lower plot: residuals (data - model) in lower plot
+
+    plt.show()
+
+    return None
+
+
+def get_fit_parameters(obj_id, time, flux, model, wave, redshift, fit_u_band=True):
+    """Get fit parameters (amplitude and Av) ...
+
+    fit_u_band (bool): If True, u-band is included in the fits TODO add False case
+    
+    Returns:
+        - The best-fit amplitude as a float
+        - The best-fit Av as a float
+    """
+        
+    # Create partial function to use `curve_fit()` with fixed parameters: obj_id, time, model, redshift, photo_data
+    # Need `partial_model_flux()` function args to be parameters that will be fit
+    # `get_model_flux()` returns OBSERVER frame flux (https://sncosmo.readthedocs.io/en/v1.1.x/models.html)
+    partial_model_flux = partial(get_model_flux, obj_id=obj_id, time=time, model=model, redshift=redshift)
+    # Fits the 2nd and 3rd parameters of `get_model_flux()` using the first parameter
+    # Note: order of args matters in `get_model_flux()`
+    if fit_u_band: 
+        fit_params, cov_matrix = curve_fit(partial_model_flux, wave, flux)
+    
+    return fit_params
+
+
+def get_model_synth_table(time, amplitude, a_v, obj_id, wave, model, redshift, spec_release, phot_release, temp_spec_tbl, spec_data):
+    """Get synthetic magnitude from modeled spectra. `time` is the first arg if `time` will be used in `curve_fit()`...
+    FIXME base curve_fit() on time?
+
+    Args:
+
+    Returns:
+
+    """
+
+    # FIXME this function is called many times
+
+    # Get flux. Flux will be used to calculate synthetic magnitude
+    # Get flux at each time. Wavelengths must be monotonically increasing
+
+    # all_flux = [get_model_flux(wave, amplitude, a_v, obj_id, t, model, redshift) for t in np.unique(time)] # time can be an array
+
+    # TODO: is it better to use lists?
```

Comment:
If you just want a container to iterate over, than it doesn't matter if you use a list.